modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPOptionCategoryServiceImpl.java

```diff
@@ -95,6 +133,23 @@ public CPOptionCategory getCPOptionCategory(long cpOptionCategoryId)
 			cpOptionCategoryId);
 	}
 
+	@Override
+	public CPOptionCategory getCPOptionCategoryByExternalReferenceCode(
+			String externalReferenceCode, long companyId)
+		throws PortalException {
+
+		CPOptionCategory cpOptionCategory =
+			cpOptionCategoryLocalService.
+				getCPOptionCategoryByExternalReferenceCode(
+					externalReferenceCode, companyId);
+
+		_cpOptionCategoryModelResourcePermission.check(
+			getPermissionChecker(), cpOptionCategory.getCPOptionCategoryId(),
+			ActionKeys.VIEW);
+
```

Comment:
I don't know if this is a problem but I think in this way you have different responses if you're using wrong ERC or you don't have permissions.
I mean...
probably I can discover which ERC exist also if I don't have the permissions

Reply:
discussed on call